src/core/place.c

```diff
@@ -76,13 +77,16 @@ northwestcmp (gconstpointer a, gconstpointer b)
     }
 
   /* probably there's a fast good-enough-guess we could use here. */
-  from_origin_a = sqrt (ax * ax + ay * ay);
-  from_origin_b = sqrt (bx * bx + by * by);
+  from_origin_a = sqrtf ((float)(ax * ax + ay * ay));
+  from_origin_b = sqrtf ((float)(bx * bx + by * by));
 
-  if (from_origin_a < from_origin_b)
+  /* from_origin_a < from_origin_b */
+  if (fabsf (from_origin_b - from_origin_a) >= FLT_EPSILON)
```

Comment:
Why?

Reply:
This is how I'm used to compare the floating point numbers on digital signal processors, also in java applications, so I think it's also correct here. Or you mean the ends, whether they are open or closed.

Reply:
I mean how this is related to warnings you are fixing in this MR?

Reply:
it's to not go to work with integers

Reply:
If you think the comparison that was there before it's simply better, I have no problem leaving the comparison as it was before.

Reply:
done
```
diff --git a/src/core/place.c b/src/core/place.c
index fa5c8e0..40370e7 100644
--- a/src/core/place.c
+++ b/src/core/place.c
@@ -80,13 +80,10 @@ northwestcmp (gconstpointer a, gconstpointer b)
   from_origin_a = sqrtf ((float)(ax * ax + ay * ay));
   from_origin_b = sqrtf ((float)(bx * bx + by * by));
 
-  /* from_origin_a < from_origin_b */
-  if (fabsf (from_origin_b - from_origin_a) >= FLT_EPSILON)
+  if (from_origin_a < from_origin_b)
     return -1;
-  /* from_origin_a > from_origin_b */
-  else if (fabsf (from_origin_a - from_origin_b) >= FLT_EPSILON)
+  else if (from_origin_a > from_origin_b)
     return 1;
-  /* from_origin_a == from_origin_b */
   else
     return 0;
 }
```

Reply:
@muktupavels recommend us not using an epsilon for comparing two floating numbers.